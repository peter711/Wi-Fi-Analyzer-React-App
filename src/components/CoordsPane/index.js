import React from "react";
import styled from "styled-components";

import { Consumer } from "../../context";

import SVGArea from "../SVGArea";
import Clients from "../SVGArea/clients";
import AccessPoint from "../SVGArea/access-point";

import generateMockClients from "../../mock/clients";

const MAX_DISTANCE_IN_M = 1000;

const CoordsPaneWrapper = styled.div`
  height: 100%;
  width: 100%;
`;

class CoordsPane extends React.Component {
  render() {
    const { radio, txPower } = this.props;
    return (
      <CoordsPaneWrapper>
        <SVGArea distance={MAX_DISTANCE_IN_M}>
          <Clients clients={generateMockClients(MAX_DISTANCE_IN_M, MAX_DISTANCE_IN_M)} />
          <AccessPoint frequency={radio} gain={txPower} />
        </SVGArea>
      </CoordsPaneWrapper>
    );
  }
}

export default () => (
  <Consumer>
    {context => <CoordsPane radio={context.radio} txPower={context.txPower} />}
  </Consumer>
);