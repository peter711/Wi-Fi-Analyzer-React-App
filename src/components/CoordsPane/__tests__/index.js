import React from "react";
import { shallow, mount } from "enzyme";

import CoordsPane from "../";

beforeEach(() => {
    jest.resetModules();
});

export const mockedContext = {
    txPower: -6,
    radio: 5,
    updateAccessPoint: jest.fn()
}

export const getCoordsPaneWithContext = (context = mockedContext) => {
    jest.doMock('../../../context', () => {
        return {
            Consumer: props => props.children(context)
        }
    });

    return require('../').default;
};

describe("CoordPane/>", () => {

    it("should render", () => {
        const CoordsPane = getCoordsPaneWithContext();
        const component = mount(<CoordsPane/>);
    });

    it("should be initialized with context", () => {
        const CoordsPane = getCoordsPaneWithContext();
        const component = shallow(<CoordsPane/>).dive();
        expect(component).toMatchSnapshot();
        expect(component.props()).toEqual({
            radio: mockedContext.radio,
            txPower: mockedContext.txPower
        });
    });

});