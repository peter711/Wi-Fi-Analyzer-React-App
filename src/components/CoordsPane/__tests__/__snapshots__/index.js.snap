// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordPane/> should be initialized with context 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Consumer>
    [Function]
  </Consumer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "radio": 5,
      "txPower": -6,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "radio": 5,
        "txPower": -6,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
